runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -679,8 +736,35 @@ private void createGameTimer(GameTimer timer)
 		infoBoxManager.addInfoBox(t);
 	}
 
+	private void createGameTimer(final GameTimer timer, final boolean replaceExisting)
+	{
+		createGameTimer(timer, replaceExisting, false);
+	}
+
+	private void createGameTimer(GameTimer timer)
+	{
+		createGameTimer(timer, true);
+	}
+
 	private void removeGameTimer(GameTimer timer)
 	{
-		infoBoxManager.removeIf(t -> t instanceof TimerTimer && ((TimerTimer) t).getTimer() == timer);
+		removeGameTimer(timer, -1);
+	}
+
+	private void removeGameTimer(GameTimer timer, long timePassedThreshold)
+	{
+		long totalTimerSeconds = timer.getDuration().getSeconds();
+		if (timePassedThreshold >= 0)
+		{
+			infoBoxManager.removeIf(
+				t -> t instanceof TimerTimer
+				&& ((TimerTimer) t).getTimer() == timer
+				&& (totalTimerSeconds - Duration.between(Instant.now(), ((TimerTimer) t).getEndTime()).getSeconds()) <= timePassedThreshold
+			);
+		}
+		else
```

Comment:
Is there any way we can remove this `if` statement? We have duplicated `t -> t instanceof TimerTimer && ((TimerTimer) t).getTimer() == timer` code as a result of it.

Reply:
We could, but that would come at an expense of computing `(totalTimerSeconds - Duration.between(Instant.now(), ((TimerTimer) t).getEndTime()).getSeconds()) <= timePassedThreshold` overtime we remove timers instead of only when a `timePassedThreshold` is defined, correct?